roles/helm_module/library/helm_cli.py

```diff
@@ -0,0 +1,190 @@
+                       'release: "{}" not found'.format(release) in err)
+
+        rc, out, err = self._run_helm(['get', release],
+                                      failed_when=failure_detect)
+        if rc != 0:
+            return
+        release_info = self._parse_helm_output(out)
+        return release_info
+
+    def install(self):
+        chart = self.get_chart()
+        release = self.params.get('release')
+        if release is None:
+            cmd = ['install', chart['name']]
+        else:
+            cmd = ['--install', release, 'upgrade', chart['name']]
+        if 'version' in chart:
+            cmd.extend(['--version', chart['version']])
+        if 'repo' in chart:
+            cmd.extend(['--repo', chart['repo']])
+        namespace = self.params.get('namespace')
+        if namespace:
+            cmd.extend(['--namespace', namespace])
+        with self.write_values_files() as values_filenames:
+            for value in values_filenames:
+                cmd.extend(['--values', value])
+            rc, out, err = self._run_helm(cmd)
+        return self._parse_helm_output(out)
+
+    def _run_helm(self, cmd, failed_when=None):
+        if failed_when is None:
+            failed_when = self._helm_std_failure
+        try:
+            rc, out, err = self.run_command([self.helm_bin] + cmd)
+        except Exception as exc:
+            raise HelmError('error running helm {} command: {}'.format(
+                ' '.join(cmd), exc))
+        else:
+            if failed_when(rc, out, err):
+                raise HelmError('Error running helm {} command (rc={}) '
+                                "out='{}' err='{}'".format(' '.join(cmd), rc, out, err))
+        return rc, out, err
+
+    @staticmethod
+    def _helm_std_failure(rc, out, err):
+        return rc != 0
+
+    def check_present(self):
+        release = self.params.get('release')
+        if release:
+            release_info_pre = self.get_release_info(release)
+        else:
+            release_info_pre = None
+
+        install = self.install()
+        release_info_post = self.get_release_info(release or install['NAME'])
+        return {'changed': release_info_pre != release_info_post,
+                'release_info_pre': release_info_pre,
+                'install': install,
+                'release_info_post': release_info_post}
+
+    def execute(self):
+        state = self.params.get('state')
+        if state == 'present':
+            return self.check_present()
+
+    @contextmanager
+    def write_values_files(self):
+        values_filenames = []
+        values_content = self.params.get('values_list', [])
+        for values in values_content:
+            temp = NamedTemporaryFile(
+                suffix='.yaml',
+                prefix='ansible.helm.values.',
+                delete=False,
+            )
+            if isinstance(values, dict):
+                values_str = json.dumps(values)
+            else:
```

Comment:
What is it if it's not a dict? A list?
What I'm getting at: shouldn't `values` be unconditionally `json.dumps`ed?